documentation/proposals/Proposal - 3.0 & 3.X Software Development Plan.md

```diff
@@ -0,0 +1,149 @@
+    - For more information, see [the Multi-Backend Input proposal](Proposal - Multi-Backend Input.md). (tenet: usability)
+
+Silk.NET 3.0 presents us with an opportunity to rethink the entire library taking into account everything we've learnt over the past 2 years of the project's development.
+
+## Development Roadmap
+
+Note that this development roadmap does not take into account unit tests, only functional tests such as experiments. The team should of course stride to add as many tests as possible where possible.
+
+### 3.0 Preview 1
+
+Before we can do anything, we need to get our brand new generators up and running. In this version:
+- The Scraper works as a minimum viable product. It has minimal support for adding extra attributes for invoking overloaders.
+    - The Khronos APIs in particular will likely be incomplete compared to 2.X in this version. 
+- The Emitter works completely as intended.
+- The Overloader works as a minimum viable product. It doesn't necessarily implement all overloads specified yet.
+- Windowing and Input are implemented for desktop platforms, and have received initial testing.
+- No development on Maths for this preview.
+- Exclusive support for .NET 6
+
+3.0 Preview 1 is not a production-ready preview and is very experimental. This preview should aim to release in early September.
+
+### 3.0 Preview 2
+
+Now that we've got an initial preview out to show what our aims are, we can start refining everything. In this version:
+- Bugfixes from 3.0 Preview 1
+- The Scraper has near-complete support for adding extra attributes for invoking overloaders.
+- The Overloader has more overloads implemented. All generic overloads should be implemented by now, but some API-specific overloads may not be implemented.
+- Android support has been restored for Windowing and Input, and have received initial testing on this platform.
+- No development on Maths for this preview.
+
+3.0 Preview 2 is not a production-ready preview and is very experimental. This preview should aim to release in late September.
+
+### 3.0 Preview 3
+
+By this preview, the groundwork has been established for 3.0 and we should ensure that all of our goals have ample progress towards the end product. In this version:
+- Bugfixes from 3.0 Preview 2
+- The Scraper is complete.
+- The Overloader is complete.
+- iOS support has been added for Windowing and Input, and have received initial testing on this platform.
+- If time permits, a start has been made on the SIMD APIs in Maths. No work has been done on integrating it into the other Maths types.
+
+3.0 Preview 3 is not a production-ready preview and is very experimental. This preview should aim to release in early October.
+
+### 3.0 Preview 4
+
+This is the first "production-ready" preview and we want users to start integrating into their workloads, so we need to make sure good progress has been made to all goals for the 3.0 update and as many forseeable breaking changes as possible done. In this version:
```

Comment:
If it's production ready it's a release candidate.

Reply:
Right, right, I think I've asked this same thing last time... I keep forgetting 😅